python/ray/serve/_private/http_proxy.py

```diff
@@ -490,24 +394,34 @@ async def __call__(self, scope, receive, send):
             scope["path"] = route_path.replace(route_prefix, "", 1)
             scope["root_path"] = root_path + route_prefix
 
+        request_id = get_random_letters(10)
         request_context_info = {
             "route": route_path,
-            "request_id": get_random_letters(10),
+            "request_id": request_id,
             "app_name": app_name,
         }
         start_time = time.time()
-        for key, value in scope["headers"]:
+        for key, value in scope.get("headers", []):
             if key.decode() == SERVE_MULTIPLEXED_MODEL_ID:
                 request_context_info["multiplexed_model_id"] = value.decode()
                 break
         ray.serve.context._serve_request_context.set(
             ray.serve.context.RequestContext(**request_context_info)
         )
-        status_code = await _send_request_to_handle(handle, scope, receive, send)
+
+        if RAY_SERVE_ENABLE_EXPERIMENTAL_STREAMING:
+            status_code = await self.send_request_to_replica_streaming(
```

Comment:
[Nit] Could you rename the following functions:

* `send_request_to_replica_streaming` -> `send_streaming_request_to_replica`
* `send_request_to_replica_unary` -> `send_unary_request_to_replica`

This way, the adjective is closer to the noun.

Reply:
It's not actually a "streaming request" vs. a "unary request," but it's the method of sending the same type of request... but that might have made it even more confusing ;)

A longer form way of saying this would be "send_request_to_replica_using_unary_protocol" or "send_request_to_replica_using_streaming_protocol"

Reply:
Oh that makes sense, thanks for explaining!